lib/ts/Arena.h

```diff
@@ -166,4 +167,34 @@ Arena::str_store(const char *str, size_t len)
   return mem;
 }
 
-#endif /* __ARENA_H__ */
+struct JemallocArena : public std::allocator<uint64_t>
```

Comment:
The "Arena" is a string alloc system ... so this is just a prefix on it.   The main reason it's here is to prevent caching/over-allocation from the intermediate system.